model/src/main/java/org/projectnessie/model/Namespace.java

```diff
@@ -33,24 +36,29 @@
 @Value.Immutable
 @JsonSerialize(as = ImmutableNamespace.class)
 @JsonDeserialize(as = ImmutableNamespace.class)
-public abstract class Namespace {
+@JsonTypeName("NAMESPACE")
+public abstract class Namespace extends Content {
```

Comment:
I think it's wrong to have the namespace elements repeated in the `Namespace` `Content` object. The name of the `Namespace` is already determined by the `ContentKey`, so it's duplicated.
Maybe just add a new `NamespaceContent` object that extends `Content` and is currenty empty?

Reply:
While I agree that the namespace is already determined in `ContentKey`, I unfortunately don't see any better way atm as we need/want to expose the `Namespace` instance in the `NamespaceApi`. 
In terms of the relationship between `ContentKey` + `Namespace` I see it this way:

Creating an actual `Namespace` `a.b.c` via the `NamespaceApi` will do a `Put.of(ContentKey.of(namespace.getElements()), namespace)`, so the full `ContentKey` name will be the same as the actual `Namespace`, namely `a.b.c`. `ContentKey` then also has a `ContentKey.getNamespace()` method, which then will return `a.b.` (omitting the last element), which makes sense, since the `Namespace` of `a.b.c` is `a.b`.